src/Servers/Kestrel/Core/src/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -486,5 +517,18 @@ private static Pipe CreateDataPipe(MemoryPool<byte> pool)
                 useSynchronizationContext: false,
                 minimumSegmentSize: pool.GetMinimumSegmentSize()
             ));
+
+        public void Dispose()
+        {
+            _disposed = true;
+
+            lock (_resetAwaitable)
+            {
+                if (_resetAwaitable.GetStatus(_resetAwaitable.Version) == ValueTaskSourceStatus.Pending)
```

Comment:
Why check?

Reply:
A stream that errored during start won't be reset